Utils/Dataflow/015_CDSDocuments/getCDSDocuments.py

```diff
@@ -34,355 +34,393 @@
-      return None
+    """ Perform CDS search by given paper info.
+
+    Returns single JSON(dict) or None
+    Search parameters:
+          aas - advanced search ("0" means no, "1" means yes).  Whether
+                search was called from within the advanced search
+                interface.
+          p1 - first pattern to search for in the advanced search
+               interface.  Much like 'p'.
+
+          f1 - first field to search within in the advanced search
+               interface.  Much like 'f'.
+
+          m1 - first matching type in the advanced search interface.
+               ("a" all of the words, "o" any of the words, "e" exact
+               phrase, "p" partial phrase, "r" regular expression).
+
+          op1 - first operator, to join the first and the second unit
+                in the advanced search interface.  ("a" add, "o" or,
+                "n" not).
+          p2 - second pattern to search for in the advanced search
+               interface.  Much like 'p'.
+
+          f2 - second field to search within in the advanced search
+               interface.  Much like 'f'.
+
+          m2 - second matching type in the advanced search interface.
+               ("a" all of the words, "o" any of the words, "e" exact
+                phrase, "p" partial phrase, "r" regular expression).
+
+          op2 - second operator, to join the second and the third unit
+                in the advanced search interface.  ("a" add, "o" or,
+                "n" not).
+          of - output format (e.g. "hb").
+          cc - current collection (e.g. "ATLAS").  The collection the
+               user started to search/browse from.
+    """
+    sys.stderr.write(paper_info["id"]+"\n")
+    #results = cds.search(cc="ATLAS", aas=1, m1="e", op1="a",
+    #                     p1=paper_info["full_title"], f1="title", m2="a",
+    #                     op2="a", p2="ARTICLE, ATLAS_Papers",
+    #                     f2="collection", m3="a", p3=paper_info["ref_code"],
+    #                     f3="report_number", of="recjson")
+    results = cds.search(cc="ATLAS", aas=1, m1="p", p1=paper_info["ref_code"],
+                         f1="reportnumber", m2="a", op2="a",
+                         p2="ARTICLE, ATLAS_Papers", f2="collection",
+                         of="recjson")
+    try:
+        res = json.loads(results)
+        if type(res) == list:
+            if len(res) > 1:
+                sys.stderr.write("(WARN) Paper search returned more than one"
+                                 " result (%s)\n"
+                                 "(WARN) Will be taken the first of the"
+                                 " list\n" % str(len(res)))
+            r = None
+            for item in res:
+                if collection_verification(item.get("collection")):
+                    r = item
+                    break
+            res = r
+        if type(res) == dict:
+            res['glance_id'] = paper_info["id"]
```

Comment:
I've just read PEP8 https://www.python.org/dev/peps/pep-0008/#string-quotes. It doesn't provide recommendations. So, I think we can use double-quotes for natural language strings and signle-quotes for technical strings.